src/make-route-builder.ts

```diff
@@ -22,9 +22,9 @@ type Suffix = `?${string}`;
 type SafePath<Path extends string> = string extends Route ? Path : Route<Path>;
 
 type ExtractPathParams<T extends string> =
-  T extends `${string}[[...${infer Param}]]${infer Rest}` ?
+  T extends `${infer Rest}[[...${infer Param}]]` ?
```

Comment:
Catch-all params are supposed to be the last one, so we 'parse' them from right to left.

Reply:
Good catch @Katona